src/util/alloc/test_alloc.c

```diff
@@ -0,0 +1,214 @@
+#include "../fd_util.h"
+
+#if FD_HAS_HOSTED && FD_HAS_X86
+
+FD_STATIC_ASSERT( FD_ALLOC_ALIGN               == 4096UL, unit_test );
+FD_STATIC_ASSERT( FD_ALLOC_FOOTPRINT           ==20480UL, unit_test );
+FD_STATIC_ASSERT( FD_ALLOC_MALLOC_ALIGN_DEFAULT==   16UL, unit_test );
+FD_STATIC_ASSERT( FD_ALLOC_JOIN_CGROUP_CNT     ==   16UL, unit_test );
+
+/* This is a torture test for same thread allocation */
+/* FIXME: IDEALLY SHOULD ADD TORTURE TEST FOR MALLOC / FREE PAIRS SPLIT
+   BETWEEN THREADS AND ADD ADD INTERPROCESS TESTING MODES. */
+
+static int    _go;
+static void * _shalloc;
+static ulong  _alloc_cnt;
+static ulong  _align_max;
+static ulong  _sz_max;
+
```

Comment:
Could main get stuck at the top?  Tests tend to be primary documentation so I want to see how to set up the test first.

Reply:
Not sure what this means.  If it got stuck, that would be bug in either the implementation or in the code.  Minimum test is just to call it without arguments on a box with a shmem sandbox and a free gigantic page on the numa node where the test will be run (much like the other tests).  Can do more extensive tests, performance, etc by passing in different sets of tile cpus, shared workspaces backed by differently backed or different affinity memory, etc.

Reply:
I guess I was asking that the code get moved up here so that you can read from the top of the file down instead of having to jump to the second to last function to see the setup...

Reply:
I thought you meant main getting stuck in loop (like the spinning waiting to go).

No strong feelings about it other than to be consistent (e.g. the existing more complex unit tests currently all have their mains at the end but lots of unit tests are simple enough structurally that main is at both the top and bottom ... the main consideration then in mind, if it is worth doing it here, it is probably worth doing it everywhere).